src/Eshopworld.WorkerProcess/Infrastructure/SystemTimer.cs

```diff
@@ -24,13 +22,13 @@ public SystemTimer()
         /// <inheritdoc />
         public async Task ExecutePeriodicallyIn(TimeSpan interval,Func<Task<TimeSpan>> executor)
         {
-            await Task.Delay(interval).ContinueWith(async t=>
+            await Task.Delay(interval,_cancellationTokenSource.Token).ContinueWith(async t=>
```

Comment:
The idea was to pass that cancelation token to the method rather than use one created internally. But if we ensure that Stop is called if the consumer of the library got cancelation token itself, that should be enough

Reply:
the cancellation is performed by calling the Stop method. It was created internally as ultimately the consumer needs a method to start the loop and one to stop it